modules/regex/regex.cpp

```diff
@@ -29,1479 +29,493 @@
-			++start;
-
-		int sub_end = s.start + s.length;
-		if (sub_end < text.length())
-			res += text.substr(sub_end, text.length() - sub_end);
-
-		p_end += res.length() - text.length();
-
-		text = res;
-	}
-	return text;
-}
-
-void RegEx::clear() {
-
-	if (root)
-		memdelete(root);
-
-	root = NULL;
-	group_names.clear();
-	lookahead_depth = 0;
-}
-
-bool RegEx::is_valid() const {
-
-	return (root != NULL);
-}
-
-String RegEx::get_pattern() const {
-
-	return pattern;
-}
-
-int RegEx::get_group_count() const {
-
-	int count = 0;
-	for (int i = 1; i < group_names.size(); ++i)
-		if (group_names[i].get_type() == Variant::INT)
-			++count;
-	return count;
-}
-
-Array RegEx::get_names() const {
-
-	Array res;
-	for (int i = 1; i < group_names.size(); ++i)
-		if (group_names[i].get_type() == Variant::STRING)
-			res.push_back(group_names[i]);
-	return res;
+	return result;
 }
 
 RegEx::RegEx() {
 
-	root = NULL;
-	lookahead_depth = 0;
+	_init_context();
 }
 
 RegEx::RegEx(const String &p_pattern) {
 
-	root = NULL;
+	_init_context();
 	compile(p_pattern);
 }
 
 RegEx::~RegEx() {
 
-	if (root)
-		memdelete(root);
-}
+	if (sizeof(CharType) == 2) {
 
-void RegExMatch::_bind_methods() {
+		if (code)
+			pcre2_code_free_16((pcre2_code_16 *)code);
+		pcre2_match_context_free_16((pcre2_match_context_16 *)match_ctx);
+		pcre2_compile_context_free_16((pcre2_compile_context_16 *)compile_ctx);
+		pcre2_general_context_free_16((pcre2_general_context_16 *)general_ctx);
 
-	ClassDB::bind_method(D_METHOD("expand", "template"), &RegExMatch::expand);
-	ClassDB::bind_method(D_METHOD("get_group_count"), &RegExMatch::get_group_count);
-	ClassDB::bind_method(D_METHOD("get_group_array"), &RegExMatch::get_group_array);
-	ClassDB::bind_method(D_METHOD("get_names"), &RegExMatch::get_names);
-	ClassDB::bind_method(D_METHOD("get_name_dict"), &RegExMatch::get_name_dict);
-	ClassDB::bind_method(D_METHOD("get_string", "name"), &RegExMatch::get_string, DEFVAL(0));
-	ClassDB::bind_method(D_METHOD("get_start", "name"), &RegExMatch::get_start, DEFVAL(0));
-	ClassDB::bind_method(D_METHOD("get_end", "name"), &RegExMatch::get_end, DEFVAL(0));
+	} else {
+
+		if (code)
+			pcre2_code_free_32((pcre2_code_32 *)code);
+		pcre2_match_context_free_32((pcre2_match_context_32 *)match_ctx);
+		pcre2_compile_context_free_32((pcre2_compile_context_32 *)compile_ctx);
+		pcre2_general_context_free_32((pcre2_general_context_32 *)general_ctx);
+	}
 }
 
 void RegEx::_bind_methods() {
 
 	ClassDB::bind_method(D_METHOD("clear"), &RegEx::clear);
 	ClassDB::bind_method(D_METHOD("compile", "pattern"), &RegEx::compile);
-	ClassDB::bind_method(D_METHOD("search:RegExMatch", "text", "start", "end"), &RegEx::search, DEFVAL(0), DEFVAL(-1));
-	ClassDB::bind_method(D_METHOD("sub", "text", "replacement", "all", "start", "end"), &RegEx::sub, DEFVAL(false), DEFVAL(0), DEFVAL(-1));
+	ClassDB::bind_method(D_METHOD("search:RegExMatch", "subject", "offset", "end"), &RegEx::search, DEFVAL(0), DEFVAL(-1));
```

Comment:
Starting today, the `:type` suffix must not be used anymore, except for virtual method binds.